tests/bluetooth/bsim_bt/test_utils/TestsuiteYamlParser.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from strictyaml import load as strictyaml_load
+from strictyaml import Map, MapPattern, Optional, Seq, Str, YAMLError
+
+LOGGER_NAME = f"bsim_plugin.{__name__.split('.')[-1]}"
+logger = logging.getLogger(LOGGER_NAME)
+
+
+class TestsuiteYamlParser:
+    bsim_config_schema = Map(
```

Comment:
Is/can this schema be available in a more human-readable form?

Reply:
Hmmm, what do you mean by human-readable form? That is how yaml schema looks like in `strictyaml` package. For example:
https://hitchdev.com/strictyaml/using/alpha/compound/fixed-length-sequences/
And I agree, that it is not as much readable as "classic" Twister's schemas (which use for yaml validation `pykwalify` package), like this one:
https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/schemas/twister/testsuite-schema.yaml

But on the other hand this yaml schema definition do not allow to prepare some "generic" schemas. If you analyze this `testsuite-schema.yaml` file, you can observe, that schema dedicated for `common` and `tests` section is duplicated. With `strictyaml` package I am able to define some part of schema just like a separate Map, and use them in several places or just split it in smaller chunks. Thanks to this `strictyaml` support more modular approach to define yaml schemas. 

Moreover `strictyaml` package offer in my opinion more robustness approach in yaml parsing:
https://hitchdev.com/strictyaml/features-removed/

But generally this is not the most important part of this plugin and if more people say that this is too complicated form of parsing yaml files, then I will change it for "classic" `pykwalify` parsing.

Reply:
By human readable I meant in a way that wasn't packed in a Python structure like `Map`. 

Something along the lines of https://gist.github.com/larslockefeer/344f2a8b508430e6689b12af8cbc0368#file-schema-yaml where it's clear which fields are available, what the types are and if they are required or optional.

Reply:
Nah, it's fine as is, but perhaps we can keep a human-readable copy of the schema somewhere then? Even if it needs to be manually created and updated. 

For documentation purposes (which we agree is a general issue with the BabbleSim tests) this would be quite helpful. 

Do you know if we have anything similar for any of our other YAML schemas in Zephyr?